src/bokeh/plotting/_figure.py

```diff
@@ -815,7 +815,7 @@ class BaseFigureOptions(Options):
         Tuple(TimeDelta, TimeDelta),
     ),
     Seq(String),
-    Object("pandas.Series"),
+    Object("narwhals.Series"),
     Object("pandas.core.groupby.GroupBy"),
```

Comment:
I think I was hoping these `Object` calls could be replaced entirely by property types that do that validation internally. IIUC we'd want a generic property type for a DataFrame series (lazy or eager, anything narhwals can hhandle) and then a `PandasGroupBy` (or similar)

Reply:
@MarcoGorelli any thoughts on this idea?

Reply:
hey, thanks for the ping! I'll try to get back to this next week